### PR TITLE
Track session ID in SessionOrchestrator

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Tracks the session ID in `SessionOrchestrator` rather than the individual session/BA services.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceSessionService.kt [hunk 1]
```diff
@@ -60,6 +41,16 @@ internal class EmbraceSessionService(
         )
     }
 
+    override fun startSessionWithManual(): String {
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceSessionService.kt:44]: `startSessionManually()`?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt [hunk 2]
```diff
@@ -9,25 +9,25 @@ internal interface SessionService {
...
      */
-    fun startSessionWithState(coldStart: Boolean, timestamp: Long)
+    fun startSessionWithState(coldStart: Boolean, timestamp: Long): String
 
     /**
-     * Ends a session in response to a state event.
+     * Starts a session manually.
      */
-    fun endSessionWithState(timestamp: Long)
+    fun startSessionWithManual(): String
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt:17]: `startSessionManually()`?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt [hunk 3]
```diff
@@ -9,25 +9,25 @@ internal interface SessionService {
...
      */
-    fun endSessionWithCrash(crashId: String)
+    fun endSessionWithState(timestamp: Long)
 
     /**
-     * Starts a session manually.
+     * Ends a session manually.
      */
-    fun startSessionWithManual()
+    fun endSessionWithManual()
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt:27]: `endSessionManually()`?

**Review by bidetofevil — APPROVED** [review]
A couple of method rename suggestions but looks good otherwise. Feel free to take or leave :-)

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt (hunk 2 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionService.kt:17]: I was on the fence about this initially. I'll leave as-is for now as I'm hoping to get to the point where there is an enum passed to the function such as `startSession(StartType)`

**Pull request merged**